astropy/units/equivalencies.py

```diff
@@ -490,23 +490,66 @@ def brightness_temperature(beam_area, disp):
         >>> fwhm_to_sigma = 1./(8*np.log(2))**0.5
         >>> beam_area = 2.*np.pi*(bmaj*bmin*fwhm_to_sigma**2)
         >>> freq = 5*u.GHz
-        >>> equiv = u.brightness_temperature(beam_area, freq)
-        >>> u.Jy.to(u.K, equivalencies=equiv)  # doctest: +FLOAT_CMP
-        217.2658703625732
+        >>> equiv = u.brightness_temperature(freq)
+        >>> (u.Jy/beam_area).to(u.K, equivalencies=equiv)  # doctest: +FLOAT_CMP
+        <Quantity 217.2658703625732 K>
+
+    Any generic surface brightness:
+
+        >>> surf_brightness = 1e6*u.MJy/u.sr
+        >>> surf_brightness.to(u.K, equivalencies=u.brightness_temperature(500*u.GHz))
+        <Quantity 130.1931904778803 K>
     """
-    beam = beam_area.to_value(si.sr)
+    if disp.unit.is_equivalent(si.sr):
+        if not beam_area.unit.is_equivalent(si.Hz):
+            raise ValueError("The inputs to `brightness_temperature` are "
+                             "frequency and angular area.")
+        warnings.warn("The inputs to `brightness_temperature` have changed. "
+                      "Frequency is now the first input, and angular area "
+                      "is the second, optional input.",
+                      DeprecationWarning)
+        disp, beam_area = beam_area, disp
+
     nu = disp.to(si.GHz, spectral())
 
-    def convert_Jy_to_K(x_jybm):
-        factor = (2 * _si.k_B * si.K * nu**2 / _si.c**2).to_value(astrophys.Jy)
-        return (x_jybm / beam / factor)
+    if beam_area is not None:
+        beam = beam_area.to_value(si.sr)
+        def convert_Jy_to_K(x_jybm):
+            factor = (2 * _si.k_B * si.K * nu**2 / _si.c**2).to(astrophys.Jy).value
+            return (x_jybm / beam / factor)
+
+        def convert_K_to_Jy(x_K):
+            factor = (astrophys.Jy / (2 * _si.k_B * nu**2 / _si.c**2)).to(si.K).value
+            return (x_K * beam / factor)
+
+        return [(astrophys.Jy, si.K, convert_Jy_to_K, convert_K_to_Jy),
+                (astrophys.Jy/astrophys.beam, si.K, convert_Jy_to_K, convert_K_to_Jy),
```

Comment:
This is a minor style point; I prefer my lists to have trailing commas much of the time, especially when they're split onto multiple lines, since it makes adding new entries easier.

That said, we have style conventions for a reason, i.e., so that we don't get a bunch of unnecessary one-character changes.  This one doesn't appear to be part of PEP.  Do we want to establish any astropy-specific guidelines?

In this case, though, I'll just drop the trailing commas because these lists should never be extended.